raizama/core/management/commands/importspreadsheet.py

```diff
@@ -0,0 +1,110 @@
+                facebook_organization = social_nets_tuple[0].replace(
+                    facebook_identifier, ""
+                )
+                facebook_organization = facebook_organization.replace(" ", "")
+                instagram_organization = social_nets_tuple[2]
+                instagram_organization = instagram_organization.replace(" ", "")
+                return facebook_organization, instagram_organization
+
+            social_nets_tuple = social_nets.rpartition(facebook_identifier)
+            instagram_organization = social_nets_tuple[0].replace(
+                instagram_identifier, ""
+            )
+            instagram_organization = instagram_organization.replace(" ", "")
+            facebook_organization = social_nets_tuple[2]
+            facebook_organization = facebook_organization.replace(" ", "")
+            return facebook_organization, instagram_organization
+
+        if match_f and not match_i:
+            facebook_organization = social_nets.replace(facebook_identifier, "")
+            facebook_organization = facebook_organization.replace(" ", "")
+            instagram_organization = ""
+            return facebook_organization, instagram_organization
+
+        if not match_f and match_i:
+            instagram_organization = social_nets.replace(instagram_identifier, "")
+            instagram_organization = instagram_organization.replace(" ", "")
+            facebook_organization = ""
+            return facebook_organization, instagram_organization
+
+        if not match_f and not match_i:
+            instagram_organization = ""
+            facebook_organization = ""
+            return facebook_organization, instagram_organization
+
+    def extract_fields(self, row_organization):
+        address_organization = row_organization.get("Endereço Completo")
+        phones_organization = row_organization["Telefone/ WhatsApp"].replace("\n", " ")
+        facebook_org, instagram_org = self.extract_social_nets(
+            row_organization["Redes Sociais"]
+        )
+
+        new_organization = Organization(
+            name=row_organization["Organização"],
+            city=row_organization["Cidade"],
+            description=row_organization["Obs"],
+            street=address_organization,
+            site=row_organization["Site"],
+            email=row_organization["E-mail"],
+            country=row_organization["País"],
+            state=row_organization["Estado"],
+            phone=phones_organization,
+            facebook=facebook_org,
+            instagram=instagram_org,
+        )
+        return new_organization
+
+    def creat_list_orgs(self, reader):
+        return [
+                self.extract_fields(row)
+                for row in reader
+                if (
+                    row["Organização"]
+                    and self.check_duplicate(row["Organização"], row["Cidade"])
+                )
+            ]
```

Comment:
Isso seria mais legível como um [gerador](https://cuducos.me/2017/08/11/geradores-do-python-o-que-sao-e-quais-problemas-resolvem.html):

```suggestion
    def organizations(self, reader):
       for row in reader:
            if not row["Organização"]:
                continue
            if self.is_duplicated(row["Organização"], row["Cidade"]):
                continue
            yield self.serialize(row)
```